napps/kytos/of_core/utils.py

```diff
@@ -0,0 +1,37 @@
+import struct
+
+from kytos.core import log
+
+from pyof.v0x01.common.header import Header
+from pyof.v0x01.common.utils import new_message_from_header
+
+
+def of_slicer(remaining_data):
+    """Slice a raw bytes into OpenFlow packets"""
+    data_len = len(remaining_data)
+    pkts = []
+    while data_len > 3:
+        length_field = struct.unpack('!H', remaining_data[2:4])[0]
+        if data_len >= length_field:
+            pkts.append(remaining_data[:length_field])
+            remaining_data = remaining_data[length_field:]
+            data_len = len(remaining_data)
+        else:
+            break
+    return pkts, remaining_data
+
+
+def unpack(packet):
+    """Unpacks the OpenFlow packet and returns a message."""
+    try:
+        header = Header()
+        header.unpack(packet[:8])
+        message = new_message_from_header(header)
+        binary_data = packet[8:]
+        if binary_data:
+            message.unpack(binary_data)
+    except Exception as e:
```

Comment:
Some comments:
1) Let's avoid getting too broad exceptions (`except Exception as e`). Here we can use `UnpackException` (It is an exception from our python-openflow library).

2) It is always better to use try/except on the minimum small number of lines:
```
header = Header()
try:
    header.unpack(packet[:8])
except UnpackException:
    log.info('Could not unpack the packet header. Packet: %s', packet)
    raise UnpackException

message = new_message_from_header(header)
binary_data = packet[8:]
if binary_data:
    try:
        message.unpack(binary_data)
    except UnpackException:
        log.info('Could not unpack the packet: %s', packet)
        raise UnpackException

return message
```

At some moment we may also get an exception from `new_message_from_header` (for now no exceptions are raised here as far as I remember and I think this may be a future improvement).

Also, that `else` was unnecessary. That return could have been done on the `try` clause of even just returned (without the else).

At last but not least, if an exception was raised, it is better to raise it to the caller instead of returning False.

@cemsbr and @beraldoleal do you agree with my comments?